### ISSUE TITLE
Glacier getJobOutput data.body is typeof String where Buffer expected?

### ISSUE THREAD

**Issue by raffi-minassian** [aws/aws-sdk-js#415]
After uploading a test 8MB binary file to glacier...

```javascript
fs.readFile('./glaciertest.mp4', function(err, buf){
  if(err){
    return console.log(err);
  }
  var checksum = glacier.computeChecksums(buf);
  var params = {
    accountId: 'XXXXXXXXXXXX',
    vaultName: 'MyTestArchive',
    archiveDescription: 'My test',
    body: buf,
    checksum: checksum.treeHash
  };
  glacier.uploadArchive(params, function(err, data) {
    if (err) console.log(err, err.stack);
    else     console.log(data);
  });
});
```

I try to download the file as follows (with job id a few hours after everything is up there the glacier way, etc) , but the file contents is not the same (now 15MB).
*data.contentType* is *application/octet-stream*, **however**
*typeof data.body* reports *string* . See below ...

```javascript
glacier.getJobOutput(params, function(err, data) {
  if (err){
    return console.log(err, err.stack);
  }
  console.log(data.contentType); // returns application/octet-stream
  console.log(typeof data.body); // returns string !?!?!
  fs.writeFile('retrievedfile.mp4', data.body, function(err){
    if(err){
      return console.log(err);
    }
    console.log('Done!');
  });
});
```

I've tried passing options with encoding as null to *fs.writeFile*, but that doesnt help. My only guess here is that *typeof data.body* should be Buffer and not String and I'm not sure how to convert it or make it return as a Buffer in the first place.

Any help, suggestions, confirmation, etc are appreciated.

**Comment by AdityaManohar** [comment]
Hey @raffi-minassian!

This is now fixed in our [v2.1.6 release](https://github.com/aws/aws-sdk-js/releases/tag/v2.1.6). Thank you for your patience!

**Comment by raffi-minassian** [comment]
Thank you for the great job @AdityaManohar !
